core/src/main/java/org/aerogear/mobile/core/metrics/MetricsService.java

```diff
@@ -60,28 +61,22 @@ public void destroy() {}
     /**
      * Send default metrics
      */
-    public void sendAppAndDeviceMetrics() {
-        if (publisher == null) {
-            throw new IllegalStateException(
-                            "Make sure you have called configure or get this instance from MobileCore.getInstance()");
-        }
-        publisher.publish(defaultMetrics);
+    public void sendAppAndDeviceMetrics(final Callback callback) {
+        this.publish(defaultMetrics, callback);
     }
 
     /**
      * Send metrics
      *
      * @param metrics Metrics to send
-     * @return this MetricsService instance
      */
-    public MetricsService publish(final Metrics... metrics) {
-        nonNull(metrics, "metrics");
+    public void publish(final Metrics[] metrics, final Callback callback) {
```

Comment:
Add another
```java
public void publish(final Metrics... metrics) {
    publish(metrics, null);
}
```

so that callers who don't need to pass the callback are not forced to pass null

Reply:
Then, there would be a method with `Metrics... metrics` and another with `Metrics[] metrics`. Would that be OK?